### DISCUSSION TITLE
更智能的AI专家翻译：根据网站信息自动生成翻译prompt

### DISCUSSION THREAD

**Discussion by akira0245** [immersive-translate/immersive-translate#1860] [Ideas]
### 新功能 | New Feature

让插件使用AI服务根据当前网页的网址、title和metadata生成更适合该网站的翻译prompt，并后续使用这个更适用于当前网站和网页的prompt（而不是通用prompt）来引导翻译

### 为什么您想要添加这些功能 | Why you want to add these features

1.现在的AI专家是个很好用的功能，但是当前大部分ai专家只是为网站匹配一个用于指导翻译风格和类别的prompt。
2.有没有可能根据网站自动生成此类prompt？或者先让ai根据网页的网址、title和metadata生成更适合该网站的prompt，之后用这个更适用于当前网站和网页的prompt来引导翻译？
3.可以有效改善因为上下文不足在翻译短句或单个多义词时候翻译错误的情况。

### 提议 | Proposal

收集当前浏览网页的元信息、域名、标题、以及可能的部分正文发送给AI，使用专用于生成翻译prompt的prompt（元prompt）来生成用于翻译的prompt，在获取到翻译prompt之后使用该prompt来进行翻译。

**Comment by theowenyoung** [comment]
谢谢反馈，这是我们正在研究的方向。 不知道您是否有更具体的建议～ 或者类似的参考文章。
